### PR TITLE
Ensure Client::login_with_sso() to be send

### PR DESCRIPTION
**Pull request by jsparber**
This removes the `info!()` log because of
https://github.com/rust-lang/rust/issues/93274

### CODE REVIEW

File: crates/matrix-sdk/src/client.rs [hunk 1]
```diff
@@ -951,7 +952,6 @@ impl Client {
         /// The number of times the SSO server will try to bind to a random port
         const SSO_SERVER_BIND_TRIES: u8 = 10;
 
-        info!("Logging in to {}", self.homeserver().await);
```
Comment (poljar) [inline crates/matrix-sdk/src/client.rs:954]: Why not move the await out of the `info!()` call?

File: crates/matrix-sdk/src/client.rs [hunk 2]
```diff
@@ -992,10 +992,9 @@ impl Client {
                 let mut n = 0u8;
                 let mut port = 0u16;
                 let mut res = Err(IoError::new(IoErrorKind::Other, ""));
-                let mut rng = thread_rng();
```
Comment (poljar) [inline crates/matrix-sdk/src/client.rs:995]: Is this somehow related? I don't see how it would be.

File: crates/matrix-sdk/src/client.rs [hunk 3]
```diff
@@ -1102,14 +1101,13 @@ impl Client {
...
     /// [`restore_login`]: #method.restore_login
     #[instrument(skip(token))]
+    #[deny(clippy::future_not_send)]
     pub async fn login_with_token(
         &self,
         token: &str,
         device_id: Option<&str>,
         initial_device_display_name: Option<&str>,
     ) -> Result<login::v3::Response> {
-        info!("Logging in to {}", self.homeserver().await);
```
Comment (poljar) [inline crates/matrix-sdk/src/client.rs:1111]: Same as the previous one.

**Review by poljar — CHANGES_REQUESTED** [review]
I don't think we want to remove those log lines.

File: crates/matrix-sdk/src/client.rs (hunk 1 above)
Comment (jsparber) [reply crates/matrix-sdk/src/client.rs:954]: added it back :)

Comment (jplatte) [reply crates/matrix-sdk/src/client.rs:954]: Yeah, this is https://github.com/tokio-rs/tracing/issues/1487 and should be fixed by simply moving out that `self.homeserver().await`.

File: crates/matrix-sdk/src/client.rs (hunk 2 above)
Comment (jsparber) [reply crates/matrix-sdk/src/client.rs:995]: yes it is, we can't hold onto [rand](https://docs.rs/rand/0.6.1/rand/index.html)::[rngs](https://docs.rs/rand/0.6.1/rand/rngs/index.html)::[ThreadRng](https://docs.rs/rand/0.6.1/rand/rngs/struct.ThreadRng.html) since it's `!Send`

**Review by poljar — APPROVED** [review]
Looks good.

**Review by jplatte — APPROVED** [review]
